src/core/core/library_utils.cpp

```diff
@@ -71,7 +71,7 @@ namespace dsn {
             }            
 # elif defined(__linux__) || defined(__FreeBSD__) || defined(__APPLE__)
             module_name = "lib" + module_name + ".so";
-            auto hmod = dlopen(module_name.c_str(), RTLD_LAZY);
+            auto hmod = dlopen(module_name.c_str(), RTLD_LAZY|RTLD_GLOBAL);
```

Comment:
Another related question is http://stackoverflow.com/questions/8302810/undefined-symbol-in-c-when-loading-a-python-shared-library, which is very similar to our situation. The "GLOBAL" solution happens to be the highest-voted answer.